### PR TITLE
transaction: Don't use callback for non-critical script errors

### LINKED ISSUE #462: dnf5 says "Error in scriptlet: return code 0"
```console
$ dnf5 -y --config=/home/zbyszek/src/systemd-stable/.mkosi.tmprvph8b_q/dnf.conf install --best --allowerasing --releasever=38 --installroot=/home/zbyszek/src/systemd-stable/.mkosi.tmprvph8b_q/root --setopt=keepcache=1 --setopt=install_weak_deps=0 '--setopt=cachedir=/home/zbyszek/src/systemd-stable/mkosi.cache/fedora~38' --setopt=reposdir= --setopt=varsdir=/home/zbyszek/src/systemd-stable/.mkosi.tmprvph8b_q/vars --setopt=check_config_file_age=0 --noplugins --nodocs acl alsa-lib bash-completion btrfs-progs compsize coreutils cryptsetup dhcp-server diffutils dnf dnsmasq dosfstools dracut e2fsprogs f2fs-tools filesystem findutils fuse gcc gdb glib2 glibc-minimal-langpack gnutls grep iproute iproute-tc kbd kernel kexec-tools kmod less libasan libbpf libcap-ng libcap-ng-utils libfido2 libgcrypt libmicrohttpd libmnl libubsan libxcrypt libxkbcommon mtools nano netcat nftables numactl-libs openssh-server openssl pam passwd polkit popt procps-ng python3 'python3dist(pefile)' 'python3dist(psutil)' 'python3dist(pytest)' qrencode quota sed strace systemd tpm2-tss tree udev util-linux util-linux valgrind vim-common wireguard-tools xfsprogs zsh
...
>>> Stop post-transaction scriptlet: kernel-core-0:6.2.10-300.fc38.x86_64
>>> Error in post-transaction scriptlet: kernel-core-0:6.2.10-300.fc38.x86_64 return code 0
>>> Running post-transaction scriptlet: kernel-modules-0:6.2.10-300.fc38.x86_64
...
```
(The middle line is highlighted in red.)
0 generally doesn't mean failure in shell scripts.

### PR DESCRIPTION
**Pull request by jan-kolarik**
When an error occurred during the scriptlet, but RPM evalutes it as non-critical, zero exit code is returned. In this case, we don't want to pass this confusing information to the user as we don't have any additional information about what happened there.

Closes #462.

### CODE REVIEW

File: libdnf/rpm/transaction.cpp [hunk 1]
```diff
@@ -579,13 +579,16 @@ void * Transaction::ts_callback(
             // total is return code - if (!RPMSCRIPT_FLAG_CRITICAL) return_code = RPMRC_OK
             auto script_type = rpm_tag_to_script_type(static_cast<rpmTag_e>(amount));
             auto nevra = trans_element_to_nevra(trans_element);
+            auto return_code = static_cast<rpmRC_e>(total);
```
Comment (j-mracek) [inline libdnf/rpm/transaction.cpp:582]: What about to use a different logger level when return_code `RPMRC_OK`. Let say logger.warning() and keep error level when return code is not OK.

**Review by j-mracek — APPROVED** [review]
LGTM

**Comment by j-mracek** [comment]
Please feel free to merge it when CI pass.

**Comment by pmatilai** [comment]
Just FWIW, I think the most appropriate thing wrt these would be a warning instead of an error or silent operation. If it failed, *something* didn't go as planned and silencing it entirely will make it hard to diagnose.

**Comment by keszybz** [comment]
> Just FWIW, I think the most appropriate thing wrt these would be a warning instead of an error or silent operation. If it failed, _something_ didn't go as planned and silencing it entirely will make it hard to diagnose.

I'm not familiar with the code, but isn't this what the patch is doing? It calls `logger.warning()`.

**Pull request merged**